### PR TITLE
[JBIDE-22358] Add load and presist of advanced properties for a connection

### PR DESCRIPTION
**Pull request by jcantrill**
Add a mechanism to load and save advanced properties of a connection

cc @adietish @fbricon

### CODE REVIEW

File: plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/connection/Connection.java [hunk 1]
```diff
@@ -88,6 +90,23 @@ public Connection(IClient client, ICredentialsPrompter credentialsPrompter, ISSL
...
 	}
 	
+	@Override
+	public Properties getExtendedProperties() {
+		return extendedProperties;
+	}
+
+	@Override
+	public void setExtendedProperty(String name, String value) {
+		Properties oldExt = new Properties(extendedProperties);
```
Comment (fbricon) [inline plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/connection/Connection.java]: I always found default properties to be tricky. For instance, considering:
```java
  Properties p1 = new Properties();
  p1.put("foo", "bar");
  Properties p2 = new Properties(p1);
```
then p1.equals(p2) return false!

Comment (jcantrill) [inline plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/connection/Connection.java]: Is this a concern for me?  I am just trying to make sure 'firePropertyChange' to actually propagates the event.  Should it really be send the original as old and then create a new one?

Comment (fbricon) [inline plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/connection/Connection.java]: my point is if your properties contain foo=bar, and you call setExtendedProperty("foo","bar"), then it'll fire a change event, but it shouldn't

File: plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/connection/Connection.java [hunk 2]
```diff
@@ -88,6 +91,23 @@ public Connection(IClient client, ICredentialsPrompter credentialsPrompter, ISSL
 		this.sslCertificateCallback = sslCertCallback;
 	}
 	
+	@Override
+	public Map<String, Object> getExtendedProperties() {
+		return new HashMap<>(extendedProperties);
+	}
+
+	@Override
+	public void setExtendedProperty(String name, String value) {
```
Comment (fbricon) [inline plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/connection/Connection.java]: Either the value is an Object or the extendedProperties map is Map<String, String>

**Comment by jcantrill** [comment]
cc @fbricon  updated to use map instead of property

File: plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/connection/IOpenShiftConnection.java [hunk 3]
```diff
@@ -43,4 +46,24 @@
...
+	 * @param ext
+	 */
+	void setExtendedProperties(Map<String, Object> ext);
+	
+	/**
+	 * Set a value of an extended property
+	 * @param name
+	 * @param value
+	 */
+	void setExtendedProperty(String name, String value);
```
Comment (fbricon) [inline plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/connection/IOpenShiftConnection.java]: value type is not consistent with the rest

**Comment by fbricon** [comment]
It doesn't seem we have tests covering the actual (de)serialization of extended properties on disk

**Comment by jcantrill** [comment]
@fbricon updated to use set(String,Object) and added save/load test

**Pull request merged**